### PR TITLE
Backend command

### PR DESCRIPTION
**Pull request by ncw**
#### What is the purpose of this change?

Add backend command framework

#### Was the change discussed in an issue or in the forum before?

Discussed in #4158

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by ncw** [comment]
@darthShadow - this is the backend command framework I was talking about.

I'd appreciate it if you'd give it the once over and tell me what you think.

See the changes to the local backend for how to implement commands for a backend.

**Comment by darthShadow** [comment]
Looks simple and clean to me. I already have a working implementation of the related service account file update based on this framework.

We can probably do with a couple more examples for using it but that can come gradually.

File: cmd/backend/backend.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    rclone backend help remote:
+    rclone backend help <backendname>
+
+Pass options to the backend command with -o. This should be key=value or key, eg:
+
+    rclone backend remote:path stats -o format=json -o long
+
+Pass arguments to the backend by placing them on the end of the line
+
+    rclone backend cleanup remote:path file1 file2 file3
```
Comment (darthShadow) [inline cmd/backend/backend.go:47]: The cleanup and remote should probably be reversed here.

File: cmd/backend/backend.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+definitions.
+
+You can discover what commands a backend understands by using
+
+    rclone backend help remote:
+    rclone backend help <backendname>
+
+Pass options to the backend command with -o. This should be key=value or key, eg:
+
+    rclone backend remote:path stats -o format=json -o long
```
Comment (darthShadow) [inline cmd/backend/backend.go:43]: The cleanup and remote should probably be reversed here.

**Comment by ncw** [comment]
Thanks for the review! I'll merge it shortly then you can build on it!

**Comment by ncw** [comment]
OK I merged this now :-)

**Pull request closed without merging**